modules/tracktion_engine/timestretch/tracktion_RubberBand.cpp

```diff
@@ -207,46 +199,44 @@ class RubberBandStretcherTests  : public UnitTest
         }
 
         // Process sin wave to time stretch and store in testBuffer
-        thisRubberBandStretcher.processData (sinBuffer.getArrayOfReadPointers(), 44100, sinBuffer.getArrayOfWritePointers());
+        rubberBandStretcher.processData (sinBuffer.getArrayOfReadPointers(), 44100, sinBuffer.getArrayOfWritePointers());
 
         AudioBuffer<float> testBuffer;
         testBuffer.setSize (2, 22050);
 
-        auto t = testBuffer.getWritePointer (0);
-        auto s = sinBuffer.getWritePointer (0);
-
-        int scount = 0;
+        int zeroCountL = 0;
+        int zeroCountR = 0;
 
-        for (int sample = 0; sample < 44100; ++sample)
+        for (int sample = 22049; sample < 44100; ++sample)
         {
-            if (s[sample] != 0 && sample >= 22050)
+            if (wL[sample] == 0)
             {
-                t[sample] = s[sample];
+                ++zeroCountL;
             }
-            else
+            if (wR[sample] == 0)
             {
-                scount++;
+                ++zeroCountR;
             }
         }
 
         // Check whether buffer has resized
-        expectEquals (scount, 22050);
-        expectLessThan (std::abs (sinBuffer.getNumSamples() - testBuffer.getNumSamples() * 2), 10);
+        expectEquals (zeroCountL, 22050);
+        expectEquals (zeroCountR, 22050);
     }
 
     //==============================================================================
     float getPitchFromNumZeroCrossings (int numZeroCrossings, int numSamples, double sampleRate)
     {
         float bufferTimeInSeconds = numSamples / sampleRate;
 
-        int numCycles = numZeroCrossings / 2;
+        float numCycles = numZeroCrossings / 2.0f;
 
         float pitchInHertz = numCycles / bufferTimeInSeconds;
 
         return pitchInHertz;
     }
 
-    int getNumZeroCrossings (AudioBuffer<float> buffer)
+    int getNumZeroCrossings (AudioBuffer<float>& buffer)
```

Comment:
This is better but it should be a `(const AudioBuffer<float>& buffer)` to show that we're not changing `buffer`